include/swift/AST/Module.h

```diff
@@ -939,6 +939,9 @@ inline SourceLoc extractNearestSourceLoc(const ModuleDecl *mod) {
   return extractNearestSourceLoc(static_cast<const Decl *>(mod));
 }
 
+/// Collects modules that this module imports via `@_exported import`.
+void collectParsedExportedImports(const ModuleDecl *M, SmallPtrSetImpl<ModuleDecl *> &Imports);
```

Comment:
I'm willing to bet that this is the cause of SR-15921. This function only works on parsed modules, i.e. ones where the files are all `SourceFile`s. The ones seen by `swift-symbolgraph-extract` (and during the build on an `-emit-module` build) are all going to be `SerializedASTFile`s, which won't work with this function. We may need to fall back to a manual decl crawl for modules which aren't parsed directly from `SourceFile`s.

Leaving this as-is may suit our purposes now, but i feel like having multiple entry points to "get a symbol graph" that have different behaviors will bite us later. If it's possible to add it to this PR, i'd love to see it.

Reply:
Hmm, interesting. I'd have to look more into that. I'd prefer if we could land this as-is and look into that issue separately as this unblocks me for what I'm trying to do.